cni/network/plugin/main.go

```diff
@@ -226,40 +227,58 @@ func main() {
 		panic("network plugin start fatal error")
 	}
 
+	defer func() {
+		netPlugin.Stop()
+
+		// release cni lock
+		if errUninit := netPlugin.Plugin.UninitializeKeyValueStore(false); errUninit != nil {
+			log.Errorf("Failed to uninitialize key-value store of network plugin, err:%v.\n", errUninit)
+		}
+
+		executionTimeMs := time.Since(startTime).Milliseconds()
+
+		if err != nil {
+			reportPluginError(reportManager, tb, err)
+			panic("network plugin execute fatal error")
```

Comment:
we're double panicking if the `netPlugin.Start` fails? 
it doesn't look like we're actually attempting to [recover from a panic](https://github.com/Azure/azure-container-networking/pull/891/files/364dc84771ed95dbe317573c940b33a4e9fda125#diff-1ac75250e969b454bbdfa034ee14417f209c876f78412186f62ae1a5be58bad0L207-L209) and it seems like this could all be accomplished by returning errors and exiting dirty